src/main/java/boost/project/LibertyServerConfigGenerator.java

```diff
@@ -0,0 +1,153 @@
+
+			// Create top level server config element
+			doc = docBuilder.newDocument();
+			Element serverRoot = doc.createElement("server");
+			serverRoot.setAttribute("description", "Liberty server generated by Liberty Boost");
+			doc.appendChild(serverRoot);
+			
+			// Create featureManager config element
+			featureManager = doc.createElement("featureManager");
+			serverRoot.appendChild(featureManager);
+			
+			// Create httpEndpoint config element with default values
+			httpEndpoint = doc.createElement("httpEndpoint");
+			httpEndpoint.setAttribute("id", "defaultHttpEndpoint");
+			httpEndpoint.setAttribute("httpPort", "9080");
+			httpEndpoint.setAttribute("httpsPort", "9443");
+			serverRoot.appendChild(httpEndpoint);
+			
+
+	  } catch (ParserConfigurationException pce) {
+		pce.printStackTrace();
+	  }
+	}
+	
+	/**
+     * Set HTTP endpoint configuration for this server config instance.
+     *
+     * @param host The host to use for this server. 
+     * @param httpPort The HTTP port to use for this server. 
+     * @param httpsPort The HTTPS port to use for this server.
+     */
+	public void setHttpEnpoints(String host, String httpPort, String httpsPort) {
+		
+		if (host != null) {
+			httpEndpoint.setAttribute("host", host);
+		}
+		if (httpPort != null) {
+			httpEndpoint.setAttribute("httpPort", httpPort);
+		}
+		if (httpsPort != null) {
+			httpEndpoint.setAttribute("httpsPort", httpsPort);
+		}
+	}
+	
+	/**
+	 * Add a Liberty feature to the server configuration
+	 *
+	 * @param featureName The full name of the Liberty feature to add. 
+     */
+	public void addFeature(String featureName) {
+
+		boolean featureAlreadyExists = false;
+		
+		NodeList existingFeatures = featureManager.getChildNodes();
+		for(int i = 0; i < existingFeatures.getLength(); i++) {
+			Node existingFeature = existingFeatures.item(i);
+			if (existingFeature.getTextContent().equals(featureName)) {
+				featureAlreadyExists = true;
+			}
+		}
+		
+		if (!featureAlreadyExists) {
+			Element feature = doc.createElement("feature");
+			feature.appendChild(doc.createTextNode(featureName));
+			featureManager.appendChild(feature);
+		}
+
+	}
+	
+	/**
+	 * Add a feature to the feature config
+	 *
+	 * @param features A list of features to add.
+	 */
+	public void addFeatures(List<String> features) {
+		
+		for(String featureName : features) {
+			addFeature(featureName);
+		}
+	}
+	
+	/**
+	 * Write the server.xml to the server config directory
+	 *
+	 * @param serverPath The full path to the server config directory.
+	 */
+	public void writeToServer(String serverPath) {
+		
+		try {
+		
+			// Replace auto-generated server.xml
+			TransformerFactory transformerFactory = TransformerFactory.newInstance();
+			Transformer transformer = transformerFactory.newTransformer();
+			transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+			transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "4");
```

Comment:
I've written server.xml generators before for another project, and I think you should add the following lines:
```       transformer.setOutputProperty(OutputKeys.METHOD, "xml");
        transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");```

Reply:
Thanks Cheryl - added.